configure.tools.sh

```diff
@@ -224,7 +224,7 @@ require_path ()
 
 require_function ()
 {
-	if check_function $1 $2
+	if check_function "$1" "$2" "-D_GNU_SOURCE"
```

Comment:
It might be better to add  this define _GNU_SOURCE to the small test program that check_function writes. Note that we add _GNU_SOURCE in ccflags_append_define anyway.